build/ris-fetch.js

```diff
@@ -0,0 +1,171 @@
+'use strict'
+
+import fs from 'node:fs';
+import {stringify} from 'qs'
+import {TMP_PATH, STADA_FILE} from './consts.js';
+
+const BBOX = { // should be approx. quadratic
+    north: 70.0,
+    west: -10.0,
+    south: 35.0,
+    east: 40.0,
+}
+
+const LIMIT_RESULTS = 10000; // maximum number of results allowed by RIS::Stations
+const PADDING_RATIO = 1.1;
+const MAX_DEPTH = 8;
+const MAX_REQUESTS = 1000;
+
+function sleep(ms) {
+    return new Promise((resolve) => {
+        setTimeout(resolve, ms);
+    });
+}
+
+const exists = async (f) => {
+    try {
+        await fs.promises.stat(f);
+        return true;
+    } catch (_) {
+        return false;
+    }
+}
+
+const requestJson = async (url, options) => {
+    try {
+        const response = await fetch(url, options);
+        if (!response.ok) {
+            throw new Error(`Failed to fetch from API: ${response.status}`);
+        }
+        return await response.json();
+    } catch (error) {
+        throw new Error(error.message);
+    }
+}
+
+const doRisStationsRequest = async (latitude, longitude, radius, groupBy) => {
+    const query = {
+        latitude,
+        longitude,
+        radius,
+        groupBy,
+        onlyActive: false,
+        limit: LIMIT_RESULTS
+    }
+    const options = {
+        headers: {
+            'DB-Client-Id': process.env.DB_CLIENT_ID,
+            'DB-Api-Key': process.env.DB_API_KEY,
+        },
+    };
+    const url = 'https://apis.deutschebahn.com/db-api-marketplace/apis/ris-stations/v1/stop-places/by-position?' + stringify(query);
+    return requestJson(url, options);
+}
+
+const fetchRisStationsInRadius = async (depth, bbox, center, radius, grouped) => {
+    const fileName = TMP_PATH + depth + '_' + Object.values(bbox).join('_') + (grouped ? '_grouped' : '_single') + '.json';
```

Comment:
Make sure to handle `TMP_PATH` values without a trailing space.

```suggestion
    const fileName = pathJoin(TMP_PATH, depth + '_' + Object.values(bbox).join('_') + (grouped ? '_grouped' : '_single') + '.json');
```